components/clp-py-utils/clp_py_utils/initialize-clp-metadata-db.py

```diff
@@ -23,88 +32,36 @@
-                    `creation_ix` INT NOT NULL,
-                    KEY `archives_creation_order` (`creator_id`,`creation_ix`) USING BTREE,
-                    UNIQUE KEY `archive_id` (`id`) USING BTREE,
-                    PRIMARY KEY (`pagination_id`)
-                )
-                """
-            )
-
-            metadata_db_cursor.execute(
-                f"""
-                CREATE TABLE IF NOT EXISTS `{table_prefix}tags` (
-                    `tag_id` INT unsigned NOT NULL AUTO_INCREMENT,
-                    `tag_name` VARCHAR(255) NOT NULL,
-                    UNIQUE KEY (`tag_name`) USING BTREE,
-                    PRIMARY KEY (`tag_id`)
-                )
-                """
-            )
-
-            metadata_db_cursor.execute(
-                f"""
-                CREATE TABLE IF NOT EXISTS `{table_prefix}archive_tags` (
-                    `archive_id` VARCHAR(64) NOT NULL,
-                    `tag_id` INT unsigned NOT NULL,
-                    PRIMARY KEY (`archive_id`,`tag_id`),
-                    FOREIGN KEY (`archive_id`) REFERENCES `{table_prefix}archives` (`id`),
-                    FOREIGN KEY (`tag_id`) REFERENCES `{table_prefix}tags` (`tag_id`)
+            if StorageEngine.CLP_S == storage_engine:
+                create_metadata_db_tables(
+                    metadata_db_cursor, table_prefix, dataset=CLP_DEFAULT_DATASET_NAME
                 )
-                """
-            )
-
-            metadata_db_cursor.execute(
-                f"""
-                CREATE TABLE IF NOT EXISTS `{table_prefix}files` (
-                    `id` VARCHAR(64) NOT NULL,
-                    `orig_file_id` VARCHAR(64) NOT NULL,
-                    `path` VARCHAR(12288) NOT NULL,
-                    `begin_timestamp` BIGINT NOT NULL,
-                    `end_timestamp` BIGINT NOT NULL,
-                    `num_uncompressed_bytes` BIGINT NOT NULL,
-                    `begin_message_ix` BIGINT NOT NULL,
-                    `num_messages` BIGINT NOT NULL,
-                    `archive_id` VARCHAR(64) NOT NULL,
-                    KEY `files_path` (path(768)) USING BTREE,
-                    KEY `files_archive_id` (`archive_id`) USING BTREE,
-                    PRIMARY KEY (`id`)
-                ) ROW_FORMAT=DYNAMIC
-                """
-            )
-
-            metadata_db_cursor.execute(
-                f"""
-                CREATE TABLE IF NOT EXISTS `{table_prefix}column_metadata_default` (
-                    `name` VARCHAR(512) NOT NULL,
-                    `type` TINYINT NOT NULL,
-                    PRIMARY KEY (`name`, `type`)
-                )
-                """
-            )
-
+                # TODO: populate the dataset table with proper default name and default path.
+                create_datasets_table(metadata_db_cursor, table_prefix)
+            # TODO: Remove the default db tables for CLP_S after the dataset feature is implemented.
```

Comment:
There are a few issues (one I should've brought up in the previous review, sorry):

* `indexer` (see compression_task.py) is using the table name `clp_column_metadata_default` rather than `clp_{dataset}_column_metadata`, so we'll need to update `indexer` to use the correct table name.
* Can we defer creating the dataset-specific tables until the next PR? I know they're harmless but logically, it makes more sense to create them when we actually have the dataset feature implemented (and it'll lead to less confusion if someone is debugging the package before the next PR).
* To ensure the column_metadata table is created even though we're not passing in a dataset name, we can make its creation unconditional (like it was before this PR); and we can name it `clp_column_metadata`.